spec/dor/indexing_service_spec.rb

```diff
@@ -0,0 +1,187 @@
+      @mock_pid = 'unique_id'
+      @mock_default_logger = double(Logger)
+      @mock_obj = double(Dor::Item)
+      @mock_solr_doc  = {id: @mock_pid, text_field_tesim: 'a field to be searched'}
+      expect(Dor::IndexingService).to receive(:default_index_logger).at_least(:once).and_return(@mock_default_logger)
+    end
+
+    it 'should reindex the object via Dor::IndexingService.reindex_pid and log success' do
+      expect(Dor).to receive(:load_instance).with(@mock_pid).and_return(@mock_obj)
+      expect(Dor::IndexingService).to receive(:reindex_object).with(@mock_obj).and_return(@mock_solr_doc)
+      expect(@mock_default_logger).to receive(:info).with("updated index for #{@mock_pid}")
+      ret_val = Dor::IndexingService.reindex_pid @mock_pid
+      expect(ret_val).to eq(@mock_solr_doc)
+    end
+
+    it 'should log the right thing if an object is not found, then re-raise the exception by default' do
+      expect(Dor).to receive(:load_instance).with(@mock_pid).and_raise(ActiveFedora::ObjectNotFoundError)
+      expect(@mock_default_logger).to receive(:warn).with("failed to update index for #{@mock_pid}, object not found in Fedora")
+      expect { Dor::IndexingService.reindex_pid(@mock_pid) }.to raise_error(ActiveFedora::ObjectNotFoundError)
+    end
+
+    it 'should log the right thing if an object is not found, but swallow the exception when should_raise_errors is false' do
+      expect(Dor).to receive(:load_instance).with(@mock_pid).and_raise(ActiveFedora::ObjectNotFoundError)
+      expect(@mock_default_logger).to receive(:warn).with("failed to update index for #{@mock_pid}, object not found in Fedora")
+      expect { Dor::IndexingService.reindex_pid(@mock_pid, nil, false) }.to_not raise_error
+    end
+
+    it "should log the right thing if there's an unexpected error, then re-raise the exception by default" do
+      unexpected_err = ZeroDivisionError.new "how'd that happen?"
+      expect(Dor).to receive(:load_instance).with(@mock_pid).and_raise(unexpected_err)
+      expect(@mock_default_logger).to receive(:warn).with(start_with("failed to update index for #{@mock_pid}, unexpected StandardError, see main app log: ["))
+      expect { Dor::IndexingService.reindex_pid(@mock_pid) }.to raise_error(unexpected_err)
+    end
+
+    it "should log the right thing if there's an unexpected Exception that's not StandardError, then re-raise the exception, even when should_raise_errors is false" do
+      stack_overflow_ex = SystemStackError.new "didn't see that one coming... maybe you shouldn't have self-referential collections?"
+      expect(Dor).to receive(:load_instance).with(@mock_pid).and_raise(stack_overflow_ex)
+      #TODO: file a bug for this expectation not working.  it seemed to work when this code was in argo, but doesn't now that it's been ported to dor-services.
+      # expect(@mock_default_logger).to receive(:error).with(start_with("failed to update index for #{@mock_pid}, unexpected Exception, see main app log: ["))
```

Comment:
this is the only thing holding me from moving this out of WIP.  does anyone know why this might've worked in argo, but not here in dor-services?  same ruby version and all.  albeit, not rails.  but from playing with it in pry, it appears that `SystemStackError` can be trapped and logged in argo, but in these dor-services tests, the code path for the general `Exception` rescue that's used to log and rethrow the error never gets hit.

@mejackreed?  @atz?